implementations/rust/ockam/ockam_node/src/context.rs

```diff
@@ -427,6 +427,27 @@ impl Context {
         Ok(child_ctx.receive::<N>().await?.take().body())
     }
 
+    /// Using a temporary new context, send a message and then receive a message with timemout
+    ///
+    /// This helper function uses [`new_detached`], [`send`], and
+    /// [`receive`] internally. See their documentation for more
+    /// details.
+    ///
+    /// [`new_detached`]: Self::new_detached
+    /// [`send`]: Self::send
+    /// [`receive`]: Self::receive
+    pub async fn send_and_receive_with_timeout<R, M, N>(&self, route: R, msg: M, timeout: u64) -> Result<N>
+    where
+        R: Into<Route>,
+        M: Message + Send + 'static,
+        N: Message,
+    {
+        let mut child_ctx = self.new_detached(Address::random_local()).await?;
+        child_ctx.send(route, msg).await?;
+        let duration = std::time::Duration::from_secs(timeout);
```

Comment:
```suggestion
        let duration = core::time::Duration::from_secs(timeout);
```

Reply:
This is why our no_std check failed https://github.com/build-trust/ockam/runs/7914079845?check_suite_focus=true#step:7:15

Reply:
I have updated it with core